public/js/Filter.js

```diff
@@ -1,23 +1,26 @@
-function SampleFilter(uniqueRow, uniqueCol, uniquePlate, uniqueGene) {
+var _ = require('lodash');
+
+function SampleFilter(sampleData) {
```

Comment:
Every function is an API, even if it is a private API! I've just glanced at JSDoc — is there any reason not to use comments structured as JSDoc expects, rather than your own syntax? It's kind of like NumPy style docstrings: even if you are not going to use the numpy docstring parser, they make the code that much more readable!